lib/samples/display_clusters/display_clusters.dart

```diff
@@ -194,3 +234,86 @@ class _DisplayClustersState extends State<DisplayClusters>
     }
   }
 }
+
+/// A simple tab that lists the GeoElements contained in the selected
+/// aggregate cluster. It shows a header with the total count, an empty
+/// state when there are none, and a scrollable list when present.
+class GeoElementsTab extends StatelessWidget {
+  const GeoElementsTab({required this.geoElements, this.onClose, super.key});
+
+  // The individual GeoElements that belong to the tapped cluster.
+  final List<GeoElement> geoElements;
+
+  // Optional close callback so this tab mirrors PopupView's "X".
+  final VoidCallback? onClose;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Padding(
+          padding: const EdgeInsets.fromLTRB(8, 8, 8, 4),
+          child: Row(
+            children: [
+              Expanded(
+                child: Text(
+                  'Geoelements',
+                  style: Theme.of(context).textTheme.titleMedium,
+                ),
+              ),
+              IconButton(
+                tooltip: 'Close',
+                icon: const Icon(Icons.close),
+                onPressed: onClose ?? () => Navigator.of(context).maybePop(),
+              ),
+            ],
+          ),
+        ),
+        const Divider(height: 1),
+
+        Padding(
+          padding: const EdgeInsets.fromLTRB(16, 16, 16, 8),
+          child: Text(
+            'Total GeoElements: ${geoElements.length}',
+            style: Theme.of(context).textTheme.titleMedium,
+          ),
+        ),
+
+        if (geoElements.isEmpty)
+          Padding(
+            padding: const EdgeInsets.fromLTRB(16, 8, 16, 16),
+            child: Text(
+              'No GeoElements found.',
+              style: Theme.of(context).textTheme.bodyMedium,
+            ),
+          )
+        else
+          Expanded(
+            child: ListView.separated(
+              padding: const EdgeInsets.fromLTRB(8, 0, 8, 16),
+              itemCount: geoElements.length,
+              separatorBuilder: (_, __) => const Divider(height: 0),
+              itemBuilder: (context, index) {
+                final name = geoElements[index].getName(index);
+                return ListTile(
+                  leading: const Icon(Icons.place_outlined),
+                  title: Text(name),
+                );
+              },
+            ),
+          ),
+      ],
+    );
+  }
+}
+
+extension GeoElementNameX on GeoElement {
```

Comment:
```suggestion
extension on GeoElement {
```
This is the pattern we follow elsewhere